rfcs/rfc-covid-agm-extension.md

```diff
@@ -0,0 +1,48 @@
+# RFC For COVID AGM date extension
+
+- Start Date: (2021-01-19)
+- Target Major Version: (None)
+- Reference Issues: (None)
+- Entity Issue: (https://app.zenhub.com/workspaces/entity-5bf2f2164b5806bc2bf60531/issues/bcgov/entity/6198)
+- Implementation PR: (None)
+
+# Summary
+
+The registrar gave all co-ops an extension on their AGMs for COVID. We need to enable them to select those extended dates in the AR with the AGM date picker. It currently restricts them to Jan 1st - Dec 31st of the calendar year.
+
+First, we need to update the default date picker validation, and allow an AGM date anywhere from Jan 1st to April 30th of the next year. Currently it is Jan 1st to Dec 31st of the same year. This is because many co-ops filing their 2020 AR hold the AGM for that year in 2021
+
+Our solution for the COVID AGM extensions is to have them check a checkbox that they are taking the blanket extension. When they check the checkbox, validation on the date picker will go from the default up to April 30th of the next year, to 6 months ahead (October 31st of the next year).
+
+ - This COVID AGM extension checkbox is optional.
+ - It's only purpose is to extend the AGM date picker validation.
+ - It will only exist in the 2020 Annual Report filing.
+ - There is no change to the ledger item if the checkbox is selected.
+ - We do not need to store if the check box was checked or not for later reference.
+
+# Basic example
+
+# Motivation
+
+# Detailed design
+
+Changes required in annual_report validation and get todo task.
+  - In validation `expected_date` has been calculated from `last_ar_date` by adding a year, which creates an issue with the new requirement
+
+    Example: Filing an ar for 2020 on 2021 Aug 15th the `last_filing_ar` will set to 2021 Aug 15th, when user tries to file an ar for the year 2021 the existing validation restrict to file on or before 2022 Aug 15th. Based on the requirement user cannot file after 2022 Apr 30. For the first time filer has same kind of issue. If the founding date is 2020 Jan 15th and trying to file on Mar 15th, the validation restrict to file after 2021 Jan 15th
+
+  - In todo task `todo_start_date` is calculating in the same logic as above (by adding 1 year). This will fail in the upcomming year.
+
+# Drawbacks
+
+# Alternatives
+
+# Adoption strategy
+
+# Unresolved questions
+
+Right now we are identifying `ARFilingYear` from `founding_date` and `last_filing_ar`. It was fine with the current implementaion (will be always in the same year on or before 31st December). Should we add another field to identify the year from filing?
```

Comment:
since agm extension will need to be implemented eventually, I think we should put something in place here that will work easily for future extensions (not just for this COVID extension). If we're adding a new field maybe it could just be something like 'ar_extension_date' or 'next_ar_deadline' which we get the year/date from. That would make it easy to apply whatever extensions are needed in the future. I haven't looked at this code in a while so if that makes things overly complicated then I'm good with your suggested solution as well

Reply:
I would like to have a discussion before deciding this

Reply:
I agree with Kials comment that a generic solution should be put in place for extensions that will allow us to do that without any code changes in the future

Reply:
@thorwolpert @kialj876 @severinbeauvais @lekshmimallika-aot

Reply:
Yes. I think so because the agm date can fall over to the next year. If I remember correctly the last ar date is set as agm date when agm is held. The todo list identifies the tasks based on last ar date and will fail. Having AR filing year separate will solve those issues